lib/models.ts

```diff
@@ -16,27 +17,30 @@ export enum DirectiveType {
     unknown = 'unknown',
 }
 
-export interface ParsedComponentsDefinition {
+export interface ParsedComponentsDefinition<T extends ComponentsDefinition> {
     components: {
         [name: string]: {
-            component: ComponentsDefinition['components'][0];
+            component: T['components'][0];
             directives: {
                 [key: string]: {
                     type: DirectiveType;
                     tag: string;    // tag name (lowercased)
                 };
             };
-            properties: ComponentsDefinition['componentProperties'][0][];
+            properties: T['componentProperties'][0][];
         };
     };
     groups: {
-        [name: string]: ComponentsDefinition['groups'][0];
+        [name: string]: T['groups'][0];
     };
     defaultComponentOnEnter: string;
-    conversionRules: ComponentsDefinition['conversionRules'];
+    conversionRules: T['conversionRules'];
 }
+export type ParsedComponentsDefinitionV10X = ParsedComponentsDefinition<ComponentsDefinitionV10X>;
+export type ParsedComponentsDefinitionV11X = ParsedComponentsDefinition<ComponentsDefinitionV11X>;
```

Comment:
quite shitty, not sure how to handle it nicely

Reply:
The type would become optional with a note in the comment that it's removed from version 1.x.x. The Digital Editor would need to support both cases anyway, since old component sets must always keep working. The types are only here to help us.